launcher-scripts/Windows-launcher.bat

```diff
@@ -1,7 +1,9 @@
 @echo off
-set OLD_JAVA_HOME=%JAVA_HOME%
-set BAT_DIRECTORY=%~dp0
-set JAVA_HOME=%BAT_DIRECTORY%jdk8u242-b08-jre
+setlocal EnableDelayedExpansion
+set "OLD_JAVA_HOME=%JAVA_HOME%"
```

Comment:
This seems irregular. Typically what I've seen is:

```suggestion
set OLD_JAVA_HOME="%JAVA_HOME%"
```

I'm pretty sure we'd only want to quote the whole if we had spaces of interest in our variable names. :thinking:

Reply:
Nope, after testing I can confirm it only works if its the entire body of the set command, it won't work otherwise.

Reply:
After some more testing I've found out that only two of the set commands require full quoting, not all of them.

```batch
set "BAT_DIRECTORY=%~dp0"
set "JAVA_HOME=%BAT_DIRECTORY%jdk8u242-b08-jre"
```

These are the two lines that require the full quoting. The other set commands are happy to have quotes after the = sign. I'll fix that up.